addons/account/models/account_account.py

```diff
@@ -432,41 +628,68 @@ def code_is_available(new_code):
 
         raise UserError(_('Cannot generate an unused account code.'))
 
+    @api.depends_context('allowed_company_ids')
     def _compute_current_balance(self):
-        balances = {
-            account.id: balance
-            for account, balance in self.env['account.move.line']._read_group(
-                domain=[('account_id', 'in', self.ids), ('parent_state', '=', 'posted')],
-                groupby=['account_id'],
-                aggregates=['balance:sum'],
-            )
+        balances = self.env['account.move.line']._read_group(
```

Comment:
Not sure why anything in this function should change

Reply:
It's because different companies that use the account might have different company currencies.
If there are several active companies, we must decide whether we want to show only the balance for the main active company, or do a currency-weighted sum. But simply summing the balances isn't correct since they might be in different currencies.

(BTW I need to write a test for this method)

Reply:
This field should display the values for only one company, and therefore only one currency

Reply:
I've changed it so that it returns the balance for the currently active company.